upscayl: init at 2.0.1

TheZombie1999 opened the pull request:
###### Description of changes

I added the upscaly appimage to nixpkgs.

This is my very first contribution so any feedback is very welcome.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

TheZombie1999 committed:
upscayl: init at 2.0.1

rapenne-s commented:
nice! I'll review it. First problem I spot is one commit is used the name "upscaly" instead of upscayl, there is also the same problem in the package name.

rapenne-s commented:
Your current version is missing a `meta` attribute with information such as homepage, license, platforms, maintainers etc...

AmeerTaweel commented:
@TheZombie1999 any updates on this?

onny commented:
Hey @TheZombie1999 thank you for this contribution! I would like to use this tool too. If you could squash the commits and change the commit title to `upscaly: init at X.X.X` I would be happy to review it :)

TheZombie1999 commented:
So i added the relevant meta attributes and updated the package to the newest version.

I am currently struggeling with the squashing because i have never done this before and don't know how deal with 
my unecessary mering of nixpkgs into my fork.

The other problem that i have is that i get some formatting errors. I tried to run nixpkgs-fmt on the two relevant files.

I seemed to have worked on the upscaly/default.nix file but not on the all-packges file . 

The problem is that it changes a lot of things that i have never touched.

Zahrun commented:
Please also rename the commit and pr to upscayl

Zahrun commented on pkgs/applications/video/upscayl/default.nix:
```diff
@@ -0,0 +1,18 @@
+{appimageTools, fetchurl}:
```
include lib
`{appimageTools, fetchurl, lib}:`

Zahrun commented:
Now this works, congratulations!
It would be nice to have a .desktop file so that upscayl also comes in the app menu.
You can see at https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/misc/protonup-qt/default.nix how they did that.

TheZombie1999 commented:
I think I implemented the .desktop file correctly now.
But I don't use them my self, therefore I can't properly test it.
Can someone pls do that ?

Zahrun commented:
I can see the .desktop entry in my app menu. However, it points to `upscayl` whereas the executable name is now `upscayl-2.0.1`. I guess you can just specify `name = upscayl` in appimageTools.wrapType2.

Zahrun reviewed (approved):
Excellent, thank you for your contribution!

Icy-Thought commented:
Also tested package by upscayling several images and it appears to be working just fine!

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/2007

SuperSandro2000 commented on pkgs/applications/video/upscayl/default.nix:
```diff
@@ -0,0 +1,35 @@
+    description = "Free and Open Source AI Image Upscaler";
+    homepage = "https://upscayl.github.io/";
+    maintainers = with maintainers; [ tim ];
+    platforms = platforms.linux ;
```
```suggestion
    platforms = platforms.linux;
```

SuperSandro2000 commented on pkgs/applications/video/upscayl/default.nix:
```diff
@@ -0,0 +1,35 @@
+appimageTools.wrapType2 rec {
+  inherit pname version src;
+
+  name = pname;
+
```
```suggestion
```
that would remove version

TheZombie1999 replied:
This is required so that the executable name is upscayl and not upscayl-2.0.1

SuperSandro2000 replied:
You want to rename the executable instead of overwriting name.

SuperSandro2000 replied:
This must be changed

SuperSandro2000 commented on pkgs/applications/video/upscayl/default.nix:
```diff
@@ -0,0 +1,35 @@
+{appimageTools, fetchurl, lib}:
+
+let 
```
```suggestion
let
```

takov751 commented:
tested the application, it works well.

drupol commented on pkgs/applications/video/upscayl/default.nix:
```diff
@@ -0,0 +1,35 @@
+  version = "2.0.1";
+  src = fetchurl {
+    url = "https://github.com/upscayl/upscayl/releases/download/v${version}/upscayl-${version}-linux.AppImage";
+    sha256 = "sha256-qsFYWUK5GwuiAaU7Y3tAM8Eea1G+R/MwoDSFUZvo/cM=";
```
```suggestion
    hash = "sha256-qsFYWUK5GwuiAaU7Y3tAM8Eea1G+R/MwoDSFUZvo/cM=";
```

Icy-Thought commented:
Might as well deprecate this PR in favor of a newer one with the [latest version of Upscayl](https://github.com/upscayl/upscayl/releases/tag/v2.5.5).

drupol commented:
Hi @TheZombie1999 ,

Do you plan to implement the changes?

If not, could you please close the PR please ?

Thanks

Icy-Thought commented:
In the meantime I am going to push a new PR with the latest update.

drupol commented:
Okay cool!

Send your PR and I'll close this one.

drupol commented:
Thank you for closing it. If you still want to be maintainer of this derivation, please chime in https://github.com/NixOS/nixpkgs/pull/237494 !
